enzyme/Enzyme/MLIR/Dialect/EnzymeOps.td

```diff
@@ -222,4 +222,11 @@ def SampleOp : Enzyme_Op<"sample",
   }];
 }
 
+def DetachGradientOp : Enzyme_Op<"detach_gradient",
```

Comment:
I’m not that big a fan of the name detach here, maybe block or stop

Reply:
Cc @vchuravy as well for name suggestions (well do the same for enzymellvm)

Reply:
- Pytorch -- `detach`
- ChainRules -- `ignore_derivatives`
- Jax -- `stop_gradient`

I prefer the chainrules one (suggests all derivatives including higher-order)

Reply:
done